internal/provider/data_source_gitlab_project_tags.go

```diff
@@ -0,0 +1,154 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mitchellh/hashstructure"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_project_tags", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_project_tags`" + ` data source allows details of a project tags.
+
+**Upstream API**: [GitLab API docs](https://docs.gitlab.com/ee/api/tags.html#list-project-repository-tags)`,
+
+		ReadContext: dataSourceGitlabProjectTagsRead,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Description: "The ID or URL-encoded path of the project owned by the authenticated user.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"order_by": {
+				Description: "Return tags ordered by `name` or `updated` fields. Default is `updated`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"sort": {
+				Description: "Return tags sorted in `asc` or `desc` order. Default is `desc`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"search": {
+				Description: "Return list of tags matching the search criteria. You can use `^term` and `term$` to find tags that begin and end with `term` respectively. No other regular expressions are supported.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"tags": {
+				Description: "List of repository tags from a project.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        dataTagsSchema,
```

Comment:
Can you refactor so that we can reuse the schema from the `gitlab_project_tag` data source?

Reply:
I also think about it. But `gitlab_project_tag` has included `project` field, but `gitlab_project_tags` doesn't.

Reply:
Yes, you could define that separate for the `gitlab_project_tag` and use `constructSchema` to combine them. Check out the `issue` / `issues` data sources ...

Reply:
Hm, okay. One moment

Reply:
I probably understand what you mean, but for this I should to rewrite tag schemes a bit. I mean also resource schemes.

Reply:
No no, let's focus on the data source here. We (or me) can do this in another PR later :)

Reply:
I had moved the scheme outside. But maybe I still hadn't caught what you mean.